string name
int32 source_id
int32[] target_ids
